src/mbgl/renderer/bucket.hpp

```diff
@@ -64,9 +47,14 @@ class Bucket : private util::noncopyable {
     bool needsUpload() const {
         return hasData() && !uploaded;
     }
+    // Returns true if this bucket fits the given layer; returns false otherwise.
+    // Implementations of this class check at least that this bucket has
+    // the same layer type with the given layer, but extra checks are also
+    // possible.
+    virtual bool fitsLayer(const style::Layer::Impl&) const { return true; }
```

Comment:
supports / accepts? Layer is parameter, `Bucket.supports(Layer)` looks bit better, imo.

Reply:
Indeed, thanks!